From 65f0c7278ec0c1cb197e3110e2bc4ebb4bd5caf4 Mon Sep 17 00:00:00 2001
From: Martin Roth <martin@coreboot.org>
Date: Sun, 9 May 2021 10:26:10 -0600
Subject: [PATCH 07/19] Makefile: Don't run genbuild_h if not doing a build

genbuild_h was being run on every make invocation - clean, distclean,
etc. to get the source date epoch value.  This value isn't used unless
a build is being done, so don't run it on non-compile make invocations.

Signed-off-by: Martin Roth <martin@coreboot.org>
Change-Id: I2afc0affc17116e0db849ea968474bc19dbb0ae1
Reviewed-on: https://review.coreboot.org/c/coreboot/+/53997
Tested-by: build bot (Jenkins) <no-reply@coreboot.org>
Reviewed-by: Angel Pons <th3fanbus@gmail.com>
Reviewed-by: Patrick Georgi <pgeorgi@google.com>
---
 Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile b/Makefile
index 8290b45e89..1e6cca44b3 100644
--- a/Makefile
+++ b/Makefile
@@ -24,7 +24,9 @@ COREBOOT_EXPORTS += top src srck obj objutil objk
 LANG:=C
 LC_ALL:=C
 TZ:=UTC0
+ifneq ($(NOCOMPILE),1)
 SOURCE_DATE_EPOCH := $(shell $(top)/util/genbuild_h/genbuild_h.sh . | sed -n 's/^.define COREBOOT_BUILD_EPOCH\>.*"\(.*\)".*/\1/p')
+endif
 # don't use COREBOOT_EXPORTS to ensure build steps outside the coreboot build system
 # are reproducible
 export LANG LC_ALL TZ SOURCE_DATE_EPOCH
-- 
2.25.1

